app/models/discussion.rb

```diff
@@ -8,10 +8,6 @@ class Discussion < ActiveRecord::Base
 
   default_scope -> { order('updated_at DESC') }
 
-  after_save :update_subscriptions
```

Comment:
Have you tried an `after_commit` callback here? It seems that touch will trigger it. 

There is also `after_touch` which would make more sense, but it apparently has [this issue](https://github.com/rails/rails/issues/8759). We can give it a go, anyway.